graphql_social_auth/decorators.py

```diff
@@ -24,6 +24,12 @@ def wrapper(cls, root, info, provider, access_token, **kwargs):
         if user is None:
             raise exceptions.GraphQLSocialAuthError(_('Invalid token'))
 
+        user_model = strategy.storage.user.user_model()
+
+        if not isinstance(user, user_model):
+            raise exceptions.GraphQLSocialAuthError(
+                _('`{}` is not a user instance').format(type(user).__name__))
```

Comment:
For this to be useful for the problem described in #7, where applications want to handle the redirect result, the exception would have to be a more specific type (such as the proposed `GraphQLSocialAuthResultIsNotUser`) and it would have to propagate the actual result object as a parameter.

Reply:
oki!, I will update the PR soon

Reply:
Just for some context, here's one way client applications can use the `GraphQLSocialAuthResultIsNotUser` exception from #7, based on how we're doing it currently for our application:

(Here `graphql_social_auth_hacks.SocialAuthMutation` is the code proposed in #7: that is, identical to the current `SocialAuthMutation`, except that it raises non-user results as `GraphQLSocialAuthResultIsNotUser`.)

```python
class SocialAuthMutation_Redirectable(graphql_social_auth_hacks.SocialAuthMutation):
    """
    Extend `SocialAuthMutation` with handling of HTTP redirects by the auth pipeline.
    """

    class Meta:
        abstract = True

    redirect_url = graphene.String(
        required=False,
        description='If set, the user should be redirected to this URL.',
    )

    @classmethod
    def mutate( cls, root, info, provider, access_token, **kwargs):
        """
        Handle `GraphQLSocialAuthResultIsNotUser` for HTTP redirects by calling `resolve_redirect`.
        """
        try:
            return super().mutate(root, info, provider, access_token, **kwargs)
        except graphql_social_auth_hacks.GraphQLSocialAuthResultIsNotUser as e:
            if isinstance(e.result, HttpResponseRedirectBase):
                redirect: HttpResponseRedirectBase = e.result
                return cls.resolve_redirect(root, info, redirect, **kwargs)
            else:
                # If the result is not an HTTP redirect, just re-raise.
                raise

    @classmethod
    def resolve_redirect( cls, root, info, redirect: HttpResponseRedirectBase, **kwargs):
        """
        Return HTTP redirects as the `redirect_url` field.

        The returned `redirect_url` field will be absolute, not relative.

        :param redirect: An auth pipeline HTTP redirect result.
        """
        redirect_url: str = site_utils.site_absolute(redirect.url)
        return cls( redirect_url=redirect_url)
```

We use this `SocialAuthMutation_Redirectable` as a base class for our redirectable `SocialAuth` and `SocialAuthJWT` mutation subclasses.

Reply:
done!, you should rename the exception to `exceptions.DoAuthError`.

Reply:
That looks like it should work great, thanks! :)